packages/ui-common/src/TxQueueContext.tsx

```diff
@@ -0,0 +1,124 @@
+// Copyright 2018-2019 @paritytech/substrate-light-ui authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { SubmittableExtrinsic, SubmittableResult } from '@polkadot/api/SubmittableExtrinsic';
+import { KeyringPair } from '@polkadot/keyring/types';
+import { RxResult } from '@polkadot/api/rx/types';
+import BN from 'bn.js';
+import { Balance } from '@polkadot/types';
+import React, { createContext, useState } from 'react';
+import { Subject } from 'rxjs';
+
+export interface PendingTransaction { // an item from the TxQueue
+  allFees: BN;
+  allTotal: BN;
+  amount: Balance;
+  id: number;
+  recipientAddress: string;
+  status: {
+    isFinalized: boolean;
+    isDropped: boolean;
+    isUsurped: boolean;
+  };
+  unsubscribe: () => void;
+}
+
+export interface SubmitParams {
+  amount: Balance;
+  allFees: BN;
+  allTotal: BN;
+  extrinsic: SubmittableExtrinsic<RxResult, RxResult>;
+  recipientAddress: string;
+  senderPair: KeyringPair;
+}
+
+interface Props {
+  children: any;
+}
+
+export const TxQueueContext = createContext({
+  txQueue: [] as PendingTransaction[],
+  submit: (params: SubmitParams) => { console.error('TxQueueContext called without Provider.'); },
+  clear: () => { console.error('TxQueueContext called without Provider.'); },
+  successObservable: new Subject(),
```

Comment:
could reduce a bit of repetition by reusing the successObservable defined after this const.